simplified-ui-accounts/src/main/java/org/nypl/simplified/ui/accounts/saml20/AccountSAML20Fragment.kt

```diff
@@ -0,0 +1,212 @@
+     */
+
+    fun create(parameters: AccountSAML20FragmentParameters): AccountSAML20Fragment {
+      val arguments = Bundle()
+      arguments.putSerializable(this.PARAMETERS_ID, parameters)
+      val fragment = AccountSAML20Fragment()
+      fragment.arguments = arguments
+      return fragment
+    }
+  }
+
+  private lateinit var profiles: ProfilesControllerType
+  private lateinit var buildConfig: BuildConfigurationServiceType
+  private lateinit var parameters: AccountSAML20FragmentParameters
+  private lateinit var progress: ProgressBar
+  private lateinit var uiThread: UIThreadServiceType
+  private lateinit var viewModel: AccountSAML20ViewModel
+  private lateinit var webView: WebView
+  private val eventSubscriptions = CompositeDisposable()
+
+  private fun constructLoginURI(): String {
+    return buildString {
+      this.append(this@AccountSAML20Fragment.parameters.authenticationDescription.authenticate)
+      this.append("&redirect_uri=")
+      this.append(AccountSAML20.callbackURI)
+    }
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    this.parameters = this.requireArguments()[PARAMETERS_ID] as AccountSAML20FragmentParameters
+
+    val services = Services.serviceDirectory()
+
+    this.uiThread =
+      services.requireService(UIThreadServiceType::class.java)
+    this.buildConfig =
+      services.requireService(BuildConfigurationServiceType::class.java)
+    this.profiles =
+      services.requireService(ProfilesControllerType::class.java)
+  }
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View? {
+    val layout = inflater.inflate(R.layout.account_saml20, container, false)
+    this.webView = layout.findViewById(R.id.saml20WebView)
+    this.progress = layout.findViewById(R.id.saml20progressBar)
+    return layout
+  }
+
+  override fun onStart() {
+    super.onStart()
+
+    this.viewModel =
+      ViewModelProviders.of(
+        this,
+        AccountSAML20ViewModelFactory(
+          profiles = this.profiles,
+          account = this.parameters.accountID,
+          description = this.parameters.authenticationDescription,
+          resources = this.resources,
+          webViewDataDir = this.requireContext().getDir("webview", Context.MODE_PRIVATE)
+        )
+      ).get(AccountSAML20ViewModel::class.java)
+
+    this.eventSubscriptions.add(
+      this.viewModel.events.subscribe(
+        this::onSAMLEvent,
+        this::onSAMLEventException,
+        this::onSAMLEventFinished
+      )
+    )
+
+    this.webView.webChromeClient = AccountSAML20ChromeClient(this.progress)
+    this.webView.webViewClient = this.viewModel.webViewClient
+    this.webView.settings.javaScriptEnabled = true
+
+    if (this.viewModel.isWebViewClientReady) {
+      this.loadLoginPage()
+    }
+  }
+
+  private fun loadLoginPage() {
+    this.webView.loadUrl(this.constructLoginURI())
```

Comment:
If you rotate the device while viewing this fragment does the webview restore its state without refetching the page?